### ISSUE TITLE
[2.4] Application#stop is not called after evolutions in development environment

### ISSUE THREAD

**Issue by tsuyoshizawa** [playframework/playframework#4682]
I noticed this problem in Play 2.4.0 with development mode.
After evolutions, my application warned below:

```
[error] c.z.h.p.HikariMBeanElf - You cannot use the same pool name for separate pool instances.
```

My `application.conf` is setting to enable Mbeans.

```
db.default.hikaricp.poolName = "play"
db.default.hikaricp.registerMbeans = true
```

The log means HikariCP couldn't release previous status.

Modules that I make are also not called stop hook.
I confirmed the behaviour by embedding `println`.

**Comment by sentolacco** [comment]
I can confirm this bug. For me this concerns the Java Play Implementation.
Is there any plan fixing this?

**Comment by schmitch** [comment]
Closed via #6437
